kyverno/kyverno
# fix: Honour generateSuccessEvents config for generating success events (#9870)

## Explanation

As of today despite the `generateSuccessEvents` flag is set to `false`, still we could see success events getting generated. This PR fixes this issue and honours the config `generateSuccessEvents` while generating success events.

## Related issue

Closes #9870


## Milestone of this PR

/milestone 1.13.0


## What type of PR is this

/kind bug


### Proof Manifests

1. Create following cluster-policy

```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: require-labels
spec:
  background: false
  validationFailureAction: Enforce
  rules:
  - name: check-for-labels
    match:
      any:
      - resources:
          kinds:
          - ConfigMap
    validate:
      message: "The label `team` is required."
      pattern:
        metadata:
          labels:
            team: "?*"
```

2. Create this "good" ConfigMap

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: good
  labels:
    dog: blue
    myservice.net/service: foo
    team: red
data:
  food: cheese
  day: monday
  color: red
```

3. See that the following event will **NOT GET GENERATED** in the default Namespace as `generateSuccessEvents` config is set to `false` by default in the `kyverno` configmap

```
$ kubectl get event
LAST SEEN   TYPE     REASON          OBJECT                         MESSAGE
3s          Normal   PolicyApplied   clusterpolicy/require-labels   ConfigMap default/good: pass
```

4. Set the config `generateSuccessEvents` to `true` in the `kyverno` configmap, restart the kyverno controller, delete the `good` configmap and repeat steps 2 to 3, and observe that now the success event is generated.  


## Checklist
